### PR TITLE
[Feature] add `standard_normal` for RewardScaling

### LINKED ISSUE #646: [Feature Request] implement a `standard_normal` keyword arg for RewardScaling transform
## Motivation

ObservationNorm transform can be called with standard normal (`(x-loc)/scale`) or regular (`x*scale+loc`). For consistency, we should implement this option for both.

I would keep it `False` by default for now (or default it to true and rename the transform to `RewardNorm`).

### PR DESCRIPTION
**Pull request by adityagandhamal**
## Description

Implement `standard_normal` kwarg for RewardScaling 

## Motivation and Context

Why is this change required? What problem does it solve?
Fixes #646 
Brings consistency to RewardScaling wrt ObservationNorm by implementing the `standard_normal` kwarg with the existing regular transform

- [ ] I have raised an issue to propose this change ([required](https://github.com/pytorch/rl/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of examples)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: torchrl/envs/transforms/transforms.py [hunk 1]
```diff
@@ -1522,8 +1531,13 @@ def __init__(
...
 
     def _apply_transform(self, reward: torch.Tensor) -> torch.Tensor:
-        reward.mul_(self.scale).add_(self.loc)
-        return reward
+        if self.standard_normal:
+            loc = self.loc
+            scale = self.scale
+            return (reward - loc) / scale
+        else:
+            reward.mul_(self.scale).add_(self.loc)
```
Comment (vmoens) [inline torchrl/envs/transforms/transforms.py:1539]: I know we were doing it like this before, but I think that in general we should avoid in-place changes (say someone has a reward computed through a model and wants to differentiate through it, things might break).
What about 
```python
reward = rewad * scale + loc
```

File: test/test_transforms.py [hunk 2]
```diff
@@ -1308,13 +1311,13 @@ def test_reward_scaling(self, batch, scale, loc, keys, device):
         td_copy = td.clone()
         reward_scaling(td)
         for key in keys_total:
-            assert (td.get(key) == td_copy.get(key).mul_(scale).add_(loc)).all()
+            if standard_normal:
+                assert (td.get(key) == ((td_copy.get(key) - loc) / scale)).all()
```
Comment (vmoens) [inline test/test_transforms.py:1315]: maybe `torch.testing.assert_close(a, b)` instead? to avoid numerical precision errors

**Review by vmoens — COMMENTED** [review]
Looks great! Can you address those two comments?

File: test/test_transforms.py (hunk 2 above)
Comment (adityagandhamal) [reply test/test_transforms.py:1315]: Got it!

**Review by vmoens — APPROVED** [review]
LGTM thanks for this!

**Pull request merged**